### PR TITLE
added the possibility to create new users via admin panel with unique and valid email adress

### PR DESCRIPTION
**Pull request by nmacuk**

### CODE REVIEW

**Comment by nmacuk** [comment]
Changed it.

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class User < ApplicationRecord
+    # Validation for unique and valid email addres
```
Comment (PiotrMisiurek) [inline app/models/user.rb]: As we discussed we can skip this commend

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+    VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i.freeze
```
Comment (PiotrMisiurek) [inline app/models/user.rb]: As we already use `Devise` (as a dependency from `ActiveAdmin`) we can use theirs email regexp `Devise.email_regexp`

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+    validates :email, uniqueness: { case_sensitive: false }, format: { with: VALID_EMAIL_REGEX }
```
Comment (PiotrMisiurek) [inline app/models/user.rb]: 👍 Props for disable case sensitive

File: db/schema.rb [hunk 4]
```diff
@@ -41,4 +41,10 @@
     t.index ["reset_password_token"], name: "index_admin_users_on_reset_password_token", unique: true
   end
 
+  create_table "users", force: :cascade do |t|
+    t.string "email"
```
Comment (PiotrMisiurek) [inline db/schema.rb]: Should we add index on email?

File: app/models/user.rb (hunk 3 above)
Comment (PiotrMisiurek) [reply app/models/user.rb]: Does it allow to have blank (nil or '') email?

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (PiotrMisiurek) [inline spec/models/user_spec.rb]: Please write test for your validations

General note: if file is generated by rails but not in use then is should be removed

File: app/models/user.rb (hunk 3 above)
Comment (nmacuk) [reply app/models/user.rb]: No it does not.

File: app/models/user.rb (hunk 2 above)
Comment (nmacuk) [reply app/models/user.rb]: Changed it

File: app/models/user.rb (hunk 1 above)
Comment (nmacuk) [reply app/models/user.rb]: Changed as well

File: db/schema.rb (hunk 4 above)
Comment (nmacuk) [reply db/schema.rb]: It's already generated and you can sort by id of email via Admin Panel

**Comment by PiotrMisiurek** [comment]
Merged

**Pull request closed without merging**